autoscale-api/src/main/java/com/sequenceiq/periscope/api/model/AutoscaleClusterHistoryResponse.java

```diff
@@ -119,27 +62,81 @@ public void setTimestamp(long timestamp) {
         this.timestamp = timestamp;
     }
 
-    public String getHostGroup() {
-        return hostGroup;
+    public ScalingActivity getScalingActivity() {
+        return scalingActivity;
     }
 
-    public void setHostGroup(String hostGroup) {
-        this.hostGroup = hostGroup;
+    public void setScalingActivity(ScalingActivity scalingActivity) {
+        this.scalingActivity = scalingActivity;
     }
 
-    public AlertType getAlertType() {
-        return alertType;
-    }
+    public static class ScalingActivity {
```

Comment:
What is the reason for the inner class?

Reply:
Autoscaling History has two types of activities, one is related to enable\disable. The other is related to upscaling\downscaling nodes and the autoscaling mode which triggered. Hence moved the second set of properties associated to optional field.

Sample structure of events.

`    {
        "stackCrn": "crn:cdp:datahub:us-west-1:smaniraju:cluster:cefd7608-3fe2-455e-8db6-b5d51be610ec",
        "scalingStatus": "SUCCESS",
        "statusReason": "Upscale successfully triggered",
        "timestamp": 1592582401507,
        "scalingActivity": {
            "adjustmentType": "EXACT",
            "adjustment": 2,
            "hostGroup": "compute",
            "originalNodeCount": 0,
            "properties": {
                "cron": "0 30 21 * * *",
                "timeZone": "Asia/Calcutta"
            },
            "autoscalingType": "SCHEDULE-BASED"
        }
    },
    {
        "stackCrn": "crn:cdp:datahub:us-west-1:smaniraju:cluster:cefd7608-3fe2-455e-8db6-b5d51be610ec",
        "scalingStatus": "ENABLED",
        "statusReason": "Autoscaling has been enabled for the cluster.",
        "timestamp": 1592582344471
    },
    {
        "stackCrn": "crn:cdp:datahub:us-west-1:smaniraju:cluster:cefd7608-3fe2-455e-8db6-b5d51be610ec",
        "scalingStatus": "DISABLED",
        "statusReason": "Autoscaling has been disabled for the cluster.",
        "timestamp": 1592580904046
    },`

Reply:
Mainly, we don't use inner classes, we separate into a different file. Could you please move to a file, if this class is not very important?